### ISSUE TITLE
Add EITC age eligibility parameters? 

### ISSUE THREAD

**Issue by MattHJensen** [OpenSourcePolicyCenter/webapp-public#225]
@martinholmer and @GoFroggyRun, two TaxBrain users have expressed a desire to modify the EITC age eligibility rules from TaxBrain. Is there any reason not to add these after https://github.com/open-source-economics/Tax-Calculator/pull/687 is merged? 

If it is fine to add them to TaxBrain, do you have a recommendation for where they should go on the static input page?

**Comment by martinholmer** [comment]
@MattHJensen  said:
> @martinholmer and @GoFroggyRun, two TaxBrain users have expressed a desire to modify the EITC age eligibility rules from TaxBrain. Is there any reason not to add these after open-source-economics/Tax-Calculator#687 is merged?

The current version of pull request #687 cannot be merged into the master branch because of code conflicts.  Those conflicts need to eliminated from #687 in a way that leaves all the unit test and validation test results unchanged.

Once that has happened, those two new policy parameters should definitely be added to TaxBrain.

> If it is fine to add them to TaxBrain, do you have a recommendation for where they should go on the static input page?

Why not add them to the bottom of the block of parameters titled `Earned Income Tax Credit` (which is the first block in the `Refundable Credits` section)?

@talumbau

**Comment by GoFroggyRun** [comment]
@MattHJensen: 

I'll be resolving the merging conflicts and updating the diagnostic table very soon.

**Comment by talumbau** [comment]
seems straightforward but in order for this to be on TaxBrain, we need to release a new version of Tax-Calculator. In order to release a new version of Tax-Calculator, we had discussed some set of regression tests to ensure that answers didn't change significantly, or if answers for certain reforms did change, they changed in a sensible way. @MattHJensen is that still the best way to proceed?

**Comment by MattHJensen** [comment]
@talumbau said: 

>In order to release a new version of Tax-Calculator, we had discussed some set of regression tests to ensure that answers didn't change significantly, or if answers for certain reforms did change, they changed in a sensible way.

Yes, I believe that is the right way to proceed. 

We already have such regression tests in Tax-Calculator:

Baseline: https://github.com/open-source-economics/Tax-Calculator/blob/master/taxcalc/tests/pufcsv_agg_expect.txt

Reforms: https://github.com/open-source-economics/Tax-Calculator/blob/master/taxcalc/comparison/reform_results.txt

We have discussed checking into the OG-USA repository output for the following reforms. I would suggest checking into the OG-USA repository Tax-Calculator results for those reforms as well (perhaps in [this]( https://github.com/open-source-economics/Tax-Calculator/blob/master/taxcalc/tests/pufcsv_agg_expect.txt) format), and any other intermediate information @jdebacker and @rickecon think would be helpful for understanding why results change over time. 


```
    reform0 = {
    2016: {
       '_II_rt1': [.09],
       '_II_rt2': [.135],
       '_II_rt3': [.225],
       '_II_rt4': [.252],
       '_II_rt5': [.297],
       '_II_rt6': [.315],
       '_II_rt7': [0.3564],
    }, }

    reform1 = {
    2016: {
        '_II_rt7': [0.35],
    }, }

    reform2 = {
    2016: {
        '_II_rt7': [0.34],
    }, }

    reform3 = {
    2016: {
        '_CG_rt3': [0.25],
    }, }

    reform4 = {
    2016: {
        '_CG_rt3': [0.24],
    }, }

    reform5 = {
    2016: {
        '_CG_rt3': [0.16],
    }, }

    reform6 = {
    2016: {
        '_STD': [ [6100*2, 12200*2, 6100*2, 8950*2, 12200*2, 6100*2, 1000*2], 
                    [6200*2, 12400*2, 6200*2, 9100*2, 12400*2, 6200*2, 1000*2],
                    [6300*2, 12600*2, 6300*2, 9250*2, 12600*2, 6300*2, 1050*2]],
    }, }  

    reform7 = {
    2016: {
        '_STD': [ [6100*2.1, 12200*2.1, 6100*2.1, 8950*2.1, 12200*2.1, 6100*2.1, 1000*2.1], 
                    [6200*2.1, 12400*2.1, 6200*2.1, 9100*2.1, 12400*2.1, 6200*2.1, 1000*2.1],
                    [6300*2.1, 12600*2.1, 6300*2.1, 9250*2.1, 12600*2.1, 6300*2.1, 1050*2.1]],
    }, }  

    reform8 = {
    2016: {
        '_II_rt3': [.15],
        '_II_rt4': [.15],
        '_II_rt5': [.15],
        '_II_brk5':[[250000, 250000, 125000, 250000, 250000, 250000]]
    }, }

    reform9 = {
    2016: {
            '_STD': [[12600, 25200, 12600, 18600, 25300, 12600, 2100]],
            '_II_brk1': [[27825, 55650, 27825, 39750, 55650, 27825]],
            '_II_brk2': [[65005, 130010, 65005, 88180, 130010, 65005]],
            '_AMT_trt1': [.0],
           '_AMT_trt2': [.0]
    },}


    reform10 = {
    2015: {
        '_II_rt5': [.3],
        '_II_rt6': [.3],
        '_II_rt7': [0.3],
    }, }
```


cc @martinholmer

**Comment by talumbau** [comment]
OK, that seems straightforward. I can the comparison for Tax-Calculator using what's in the `comparison/` directory. I will also run Tax-Calculator 0.6.2 + OGUSA 0.5.3 (current TaxBrain settings) against Tax-Calculator master and OGUSA master with the 10 reforms above and see if there are differences.

**Comment by talumbau** [comment]
I ran the above 11 reforms on TaxBrain and did an OGUSA run with default parameters from each. For each reform above, here is the microsim and resulting output page for the OGUSA run:

reform_8 
http://www.ospc.org/taxbrain/3305/
http://www.ospc.org/dynamic/results/122/


reform_9
http://www.ospc.org/taxbrain/3306/
http://www.ospc.org/dynamic/results/123/


reform_4
http://www.ospc.org/taxbrain/3301/
http://www.ospc.org/dynamic/results/124/

reform_5
http://www.ospc.org/taxbrain/3302/
http://www.ospc.org/dynamic/results/125/

reform_6
http://www.ospc.org/taxbrain/3303/
http://www.ospc.org/dynamic/results/127/

reform_7
http://www.ospc.org/taxbrain/3304/
http://www.ospc.org/dynamic/results/126/


reform_0
http://www.ospc.org/taxbrain/3297/
http://www.ospc.org/dynamic/results/129/

reform_1
http://www.ospc.org/taxbrain/3298/
http://www.ospc.org/dynamic/results/128/


reform_2
http://www.ospc.org/taxbrain/3299/
http://www.ospc.org/dynamic/results/131/

reform_3
http://www.ospc.org/taxbrain/3300/
http://www.ospc.org/dynamic/results/130/


reform_10
http://www.ospc.org/taxbrain/3307/
http://www.ospc.org/dynamic/results/132/

I also ran these 11 reforms with the latest main of Tax-Calculator and [this branch](https://github.com/open-source-economics/OG-USA/pull/216) of OGUSA as a comparison but I think we are waiting on one more feature to get to Tax-Calculator master before final comparisons and release. cc @MattHJensen

**Comment by MattHJensen** [comment]
@talumbau, is there a way to have a script in the OG-USA repo that generates these results and checks them into the repo for every new release of taxcalc and OG-USA?  Tax-calculator contributors who have access to the puf run such a script before every PR: https://github.com/open-source-economics/Tax-Calculator/blob/master/taxcalc/tests/test_pufcsv.py#L34.

**Comment by talumbau** [comment]
Yes, there's a way to do this, but if we just have a script that runs everything locally, that would be less than ideal because it would be about 35-40 hours of compute time on one machine. We might want to have a script that can call out to some worker nodes, and then retrieve results when the work is done.

**Comment by talumbau** [comment]
Also, there is a mistake above and reforms 6 and 7 are wrong. The microsim listed is not a TaxBrain run for the indicated reform. We can't currently run reforms 6 and 7 on TaxBrain until the merge of https://github.com/open-source-economics/Tax-Calculator/pull/719

**Comment by MattHJensen** [comment]
> Yes, there's a way to do this, but if we just have a script that runs everything locally, that would be less than ideal because it would be about 35-40 hours of compute time on one machine. We might want to have a script that can call out to some worker nodes, and then retrieve results when the work is done.

This makes sense to me as well, but I'd rather not wait for the new TaxBrain release until that is ready.

**Comment by talumbau** [comment]
<img width="866" alt="screen shot 2016-05-05 at 9 42 42 am" src="https://cloud.githubusercontent.com/assets/1891418/15046519/c2ce4cbe-12a5-11e6-9024-c2b251053777.png">

this is what the new section will look like. cc @MattHJensen @martinholmer

**Comment by talumbau** [comment]
If I can a sign off on the above screen cap then I think we are good to merge #230

**Comment by MattHJensen** [comment]
@martinholmer and @GoFroggyRun could you please review?

**Comment by martinholmer** [comment]
It's maximum and minimum, not minimal and maximal.

**Comment by MattHJensen** [comment]
These are drawn directly form tax-calculator master long_name. I will open a PR to tax-calculator to fix this. @martinholmer @GoFroggyRun

**Comment by talumbau** [comment]
Does that mean we should post a PR to Tax-Calculator for a fix? The text is taken from current_law_policy.json:

https://github.com/open-source-economics/Tax-Calculator/blob/master/taxcalc/current_law_policy.json#L372

https://github.com/open-source-economics/Tax-Calculator/blob/master/taxcalc/current_law_policy.json#L386

I do not mean that I copied the text from there, but rather that we populate the field based on the text in the JSON file.

**Comment by MattHJensen** [comment]
see https://github.com/open-source-economics/Tax-Calculator/pull/727. Waiting for tests to pass and then merging.

**Comment by MattHJensen** [comment]
All set @talumbau

**Comment by martinholmer** [comment]
T.J.,

Now that I'm back in the office, I don't see where on the TaxBrain page these new EITC parameters are going to be.  The title is completely unnecessary if they are placed where I suggested: at the bottom of the EITC parameters.  Where are you putting them?

The minimal and maximal wording is from Sean's pull request; sorry I missed fixing that.

@talumbau @MattHJensen

**Comment by talumbau** [comment]
Yes I misunderstood. Here is what it looks like now with the newly released 0.6.3:

<img width="899" alt="screen shot 2016-05-05 at 12 39 24 pm" src="https://cloud.githubusercontent.com/assets/1891418/15051501/8d4d0558-12be-11e6-94a3-157e63fd329a.png">

The fields are now the last two in the Earned Income Tax Credit block

**Comment by martinholmer** [comment]
T.J. (@talumbau) said:
> The fields are now the last two in the Earned Income Tax Credit block.

Looks great!

**Comment by MattHJensen** [comment]
Lgtm too.

On May 5, 2016, at 1:43 PM, Martin Holmer <notifications@github.com<mailto:notifications@github.com>> wrote:


T.J. (@talumbau<https://github.com/talumbau>) said:

The fields are now the last two in the Earned Income Tax Credit block.

Looks great!
